examples/store-hash-query/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"net"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/waku-org/go-waku/waku/v2/node"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"github.com/waku-org/go-waku/waku/v2/protocol/store"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var log = utils.Logger().Named("store-hash-query")
+
+func randomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+func main() {
+	ctx := context.Background()
+
+	key, err := randomHex(32)
+	if err != nil {
+		log.Error("Could not generate random key", zap.Error(err))
+		return
+	}
+	prvKey, err := crypto.HexToECDSA(key)
+	if err != nil {
+		log.Error("Could not convert hex into ecdsa key", zap.Error(err))
+		return
+	}
+	hostAddr, _ := net.ResolveTCPAddr("tcp", "0.0.0.0:0")
+	wakuNode, err := node.New(
+		node.WithPrivateKey(prvKey),
+		node.WithHostAddress(hostAddr),
+		node.WithNTP(),
+		node.WithWakuStore(),
```

Comment:
Doesn't this mean localNode is also acting as storeNode ? 

As per https://github.com/waku-org/go-waku/blob/269417c5e979aeec92d323bed280d23da95379ea/waku/v2/node/wakunode2.go#L197 it seems we take this flag to enable Store in ENR bitfield which indicates local node is a store node.